lib/pure/collections/queues.nim

```diff
@@ -8,56 +8,168 @@
 #
 
 ## Implementation of a `queue`:idx:. The underlying implementation uses a ``seq``.
+##
+## None of the procs that get an individual value from the queue can be used
+## on an empty queue.
+## If compiled with `boundChecks` option, those procs will raise an `IndexError`
+## on such access. This should not be relied upon, as `-d:release` will
+## disable those checks and may return garbage or crash the program.
+##
+## As such, a check to see if the queue is empty is needed before any
+## access, unless your program logic guarantees it indirectly.
+##
+## .. code-block:: Nim
+##   proc foo(a, b: Positive) =  # assume random positive values for `a` and `b`
+##     var q = initQueue[int]()  # initializes the object
+##     for i in 1 ..< a: q.add i  # populates the queue
+##
+##     if b < q.len:  # checking before indexed access
+##       echo "The element at index position ", b, " is ", q[b]
+##
+##     # The following two lines don't need any checking on access due to the
+##     # logic of the program, but that would not be the case if `a` could be 0.
+##     assert q.front == 1
+##     assert q.back == a
+##
+##     while q.len > 0:  # checking if the queue is empty
+##       echo q.pop()
+##
+##
 ## Note: For inter thread communication use
 ## a `Channel <channels.html>`_ instead.
 
+proc englishOrdinal(n: SomeInteger): string =
+  # Temporary proc. Needs to be moved somewhere else as it can be reused in
+  # other places too.
+  # If this accepted number strings instead and only gave out the letters it
```

Comment:
Since it's already a generic (SomeInteger is a generic type class) and $ on strings is a nop, you only need to remove the 'n: SomeInteger' restriction and then it will work with anything supporting $